domains/runtime/evm/src/lib.rs

```diff
@@ -156,6 +168,21 @@ impl fp_self_contained::SelfContainedCall for RuntimeCall {
     ) -> Option<Result<(), TransactionValidityError>> {
         match self {
             RuntimeCall::Ethereum(call) => {
+                // Withdraw the consensus chain storage fee from the caller and record
+                // it in the `BlockFees`
+                let consensus_storage_fee =
+                    BlockFees::consensus_chain_byte_fee() * Balance::from(len as u32);
+                match <InnerEVMCurrencyAdapter as pallet_evm::OnChargeEVMTransaction<Runtime>>::withdraw_fee(
+                    info,
+                    consensus_storage_fee.into(),
+                ) {
+                    Ok(None) => {}
+                    Ok(Some(paid_consensus_storage_fee)) => {
+                        BlockFees::note_consensus_storage_fee(paid_consensus_storage_fee.peek())
+                    }
```

Comment:
I was looking around the how evm charges the fee and it seems it will also change the storage fee per byte of data as gas. I think we need to be sure about this else we might be charging users double the amount of storage fees

Reply:
Not sure what you mean by `charging the same as substrate`
EVM already charges the user for extrinsic storage as part of the `intrinsic_gas`. This is mentioned as part of the Ethereum yellow paper here - https://ethereum.github.io/yellowpaper/paper.pdf

Specifically, you need to look at page 9 `6.2 Execution` that talks about how intrinsic gas `g_0` is constructed which also include storage fees. So essentially we are charging user twice.

I believe this is the reason why frontier do not use pre_dispatch and instead only use post_dispatch

Reply:
Yeah, that is what I meant previously by "Evm chain has its own model for the tx fee, the way I understand is the storage cost is already factored into the gas, so frontier only charges for used_gas * gas_price".

The gas already accounts for the storage fee, compute fee and tip. But similar to substrate transactions we also need to charge an additional consensus chain storage fee.

Reply:
ahh I see. I overlooked this